File: 프로그래머스/Level2/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    int num;
    
    for(int j=1; j<=n; j++){
        for(int i=j; i<=n; i++){
            num += i;
            if(num == n){
                answer ++;
                break;
            }
            else if(num > n){
                break;
            }
        }
        num = 0;
    }
    
    return answer;
}

File: 프로그래머스/Level2/숫자의 표현.js
function solution(n) {
    let answer = 1;
    
    for(let i=1; i<=n-1; i++){
        let sum = i;
        
        for(let j=i+1; j<=n; j++){
            if(sum === n){
                answer ++;
                break;
            }
            else if(sum > n) break;
            
            sum += j;
        }
    }
    return answer;
}
